Networking/Networking/Model/Product/Product.swift

```diff
@@ -266,17 +266,18 @@ public struct Product: Codable {
 
         let regularPrice = try container.decodeIfPresent(String.self, forKey: .regularPrice)
 
+        let onSale = try container.decode(Bool.self, forKey: .onSale)
+
         // Even though a plain install of WooCommerce Core provides string values,
         // some plugins alter the field value from String to Int or Decimal.
         var salePrice = ""
```

Comment:
Now I wonder if plugins could return `null` for the `sale_price`:

```json
            "sale_price": null,
```

I tried it out with your unit test and it looks like the `salePrice` will be **an empty string** because none of the `if let` conditions would be met so no parsing ever happens. 

This is probably not gonna happen but it's probably better to be safe. We can fix this by changing the default to `"0"`:

```suggestion
        var salePrice = "0"
```

Side note: It's really great that we have that unit test because I can play around with these scenarios. 😄

Reply:
If I understand correctly what you mean, I think it's ok to be an empty string, in case of ` "sale_price": null,`, because `0` is a valid value, so you are saying that the product has a sale price (free, but it's not).

Reply:
Gah! You're right. Sorry, my bad.

Reply:
Don't worry 😉 I request your final review.